android/engine/src/main/java/org/smartregister/fhircore/engine/data/local/register/RegisterRepository.kt

```diff
@@ -397,21 +410,77 @@ constructor(
     val relatedResourcesConfig = resourceConfig.relatedResources
     val baseResourceClass = baseResourceConfig.resource.resourceClassType()
     val baseResourceType = baseResourceClass.newInstance().resourceType
+    val secondaryResourceConfig = profileConfiguration.secondaryResources
 
     val baseResource: Resource =
       withContext(dispatcherProvider.io()) {
         fhirEngine.get(baseResourceType, resourceId.extractLogicalIdUuid())
       }
 
-    return retrieveRelatedResources(
-      relatedResourcesConfig = relatedResourcesConfig,
+    val relatedResources = LinkedList<RelatedResourceData>()
+
+    relatedResourcesConfig.forEach { config: ResourceConfig ->
+      val resources =
+        withContext(dispatcherProvider.io()) {
+          searchRelatedResources(
+            resourceConfig = config,
+            baseResourceType = baseResourceType,
+            baseResource = baseResource,
+            fhirPathExpression = config.fhirPathExpression
+          )
+        }
+      relatedResources.addAll(resources)
+    }
+
+    if (!secondaryResourceConfig.isNullOrEmpty()) {
+      relatedResources.addAll(retrieveSecondaryResources(secondaryResourceConfig))
+    }
+
+    return processResourceData(
+      relatedResources = relatedResources,
       baseResourceType = baseResourceType,
       baseResource = baseResource,
       views = profileConfiguration.views,
       rules = profileConfiguration.rules
     )
   }
 
+  private suspend fun retrieveSecondaryResources(
```

Comment:
This function is duplicating the implementation in [RegisterRepository.kt#L93](https://github.com/opensrp/fhircore/blob/31a064bca0d1874d8dcf1ec72ea4d3045fbde1db/android/engine/src/main/java/org/smartregister/fhircore/engine/data/local/register/RegisterRepository.kt#L93)

Since loading data for the secondary resources is more like loading data for base resource, the implementation can be refactored and used in both cases.